pkg/registry/client.go

```diff
@@ -80,91 +88,51 @@ func NewClient(options ...ClientOption) (*Client, error) {
 	if client.credentialsFile == "" {
 		client.credentialsFile = helmpath.ConfigPath(CredentialsFileBasename)
 	}
-	if client.authorizer == nil {
-		authClient, err := dockerauth.NewClientWithDockerFallback(client.credentialsFile)
-		if err != nil {
-			return nil, err
+	if client.httpClient == nil {
+		// taken from http.DefaultTransport, copied here not to interfere with
+		// it
+		dialer := net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
 		}
-		client.authorizer = authClient
-	}
-
-	resolverFn := client.resolver // copy for avoiding recursive call
-	client.resolver = func(ref registry.Reference) (remotes.Resolver, error) {
-		if resolverFn != nil {
-			// validate if the resolverFn returns a valid resolver
-			if resolver, err := resolverFn(ref); resolver != nil && err == nil {
-				return resolver, nil
-			}
+		transport := http.Transport{
+			Proxy:                 http.ProxyFromEnvironment,
+			DialContext:           dialer.DialContext,
+			ForceAttemptHTTP2:     true,
+			MaxIdleConns:          100,
+			IdleConnTimeout:       90 * time.Second,
+			TLSHandshakeTimeout:   10 * time.Second,
+			ExpectContinueTimeout: 1 * time.Second,
 		}
 
-		headers := http.Header{}
-		headers.Set("User-Agent", version.GetUserAgent())
-		dockerClient, ok := client.authorizer.(*dockerauth.Client)
-		if ok {
-			username, password, err := dockerClient.Credential(ref.Registry)
-			if err != nil {
-				return nil, errors.New("unable to retrieve credentials")
-			}
-			// A blank returned username and password value is a bearer token
-			if username == "" && password != "" {
-				headers.Set("Authorization", fmt.Sprintf("Bearer %s", password))
-			} else {
-				headers.Set("Authorization", fmt.Sprintf("Basic %s", basicAuth(username, password)))
-			}
+		client.httpClient = &http.Client{
+			Transport: retry.NewTransport(&transport),
 		}
+	}
+	if client.authorizer == nil {
+		header := http.Header{}
+		header.Add("User-Agent", version.GetUserAgent())
 
-		opts := []auth.ResolverOption{auth.WithResolverHeaders(headers)}
-		if client.httpClient != nil {
-			opts = append(opts, auth.WithResolverClient(client.httpClient))
-		}
-		if client.plainHTTP {
-			opts = append(opts, auth.WithResolverPlainHTTP())
+		authorizer := auth.Client{
+			Header: header,
+			Client: client.httpClient,
 		}
-		resolver, err := client.authorizer.ResolverWithOpts(opts...)
+
+		store, err := credentials.NewFileStore(client.credentialsFile)
```

Comment:
Thanks, just noting that I also needed to enable plaintext put for the unit tests to pass

Reply:
BTW in case you want to read docker credentials as a fallback, see [example](https://pkg.go.dev/github.com/oras-project/oras-credentials-go#example-NewStoreWithFallbacks-ConfigAsPrimaryStoreDockerAsFallback).